### PR TITLE
Vince revisions

### PR DESCRIPTION
**Pull request by vas235**
This adds the User class and implements it into the Repository class so now when a repository is requested and all the pulls have been gathered, data is collected for all users that made those pulls and how many pull each user makes, this allowed me to also wrap up the last display function (display_users_per_repo) for the AllRepositories class.

We should now be able to create display functions for the user data of each repository. if the user data does not have all the correct fields vova wants we can adjust it.

### CODE REVIEW

File: gitdata.py [hunk 1]
```diff
@@ -14,6 +108,7 @@ def __init__(self, login):
 
     def __str__(self):
         return f"{self.login}"
+>>>>>>> main
```
Comment (j-tenny) [inline gitdata.py:111]: These text markers get added to the .py files when you do merge operations. You must delete these lines from the code or else it won't run.

File: gitdata.py [hunk 2]
```diff
@@ -65,10 +162,60 @@ def pull_requests_to_pandas(self):
...
+        return users_json
+
+    def get_users(self,token=None):
+        # Temporarily create an empty list
+        user_list = list()
+        
+        #iterate across pull requests
+        for pull in self.pull_requests:
+            #check if name in user list
+            check = [i for i,d in enumerate(user_list) if d['name'] == pull.user]
```
Comment (j-tenny) [inline gitdata.py:183]: Could have just used a set instead of a list to avoid duplicates

Comment (vas235) [reply gitdata.py:183]: good call, that will be cleaner, I can make that change

**Review by j-tenny — APPROVED** [review]

File: gitdata.py (hunk 1 above)
Comment (vas235) [reply gitdata.py:111]: I see, I will remove those in the future

**Pull request merged**